app/src/main/java/com/example/blah/mobilestudio/Activities/SafeActivity.java

```diff
@@ -0,0 +1,157 @@
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityCreate(savedInstanceState);
+        } catch (Throwable th){
+            Log.e("On create error", th.getMessage());
+        }
+
+    }
+
+    @Override
+    protected final void onResume(){
+        super.onResume();
+        try {
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityResume();
+        } catch (Throwable th){
+            Log.e("On resume error", th.getMessage());
+        }
+    }
+
+    @Override
+    protected final void onStart(){
+        super.onStart();
+        try {
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityStart();
+        } catch (Throwable th){
+            Log.e("On start error", th.getMessage());
+        }
+    }
+
+    @Override
+    protected final void onRestart(){
+        super.onRestart();
+        try {
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityRestart();
+        } catch (Throwable th){
+            Log.e("On restart error", th.getMessage());
+        }
+    }
+
+    @Override
+    protected final void onPause(){
+        super.onPause();
+        try {
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityPause();
+        } catch (Throwable th){
+            Log.e("On pause error", th.getMessage());
+        }
+    }
+
+    @Override
+    protected final void onStop(){
+        super.onStop();
+        try {
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityStop();
+        } catch (Throwable th){
+            Log.e("On stop error", th.getMessage());
+        }
+    }
+
+    @Override
+    protected final void onDestroy(){
+        super.onDestroy();
+        try {
+            // Common logic to all classes
+        } catch (Throwable th){
+            commonLogicError = th;
+            Log.e("Common logic error", commonLogicError.getMessage());
+        } try {
+            onActivityDestroy();
+        } catch (Throwable th){
+            Log.e("On destroy error", th.getMessage());
+        }
+    }
+
+
+
```

Comment:
I know these lines are over commented, but it's for the Javadoc. It is there so the developer will know which method is which, without having to refer to the code in this class.